src/common/armor.cc

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Liu Changcheng <changcheng.liu@aliyun.com>
+ * Author: Liu Changcheng <changcheng.liu@aliyun.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "armor.h"
+#include "base64/base64_plain.h"
+#include "base64/probe_arch.h"
+#include "base64/intel_simd.h"
+#include "base64/avx2/base64_intel_avx2.h"
+
+/* choose best implementation based on the CPU architecture.  */
+base64_encode_func_t choose_encode_base64(void) {
```

Comment:
the indent does not comply to our code style.

Reply:
@tchaikov 
Follow the below style, right?
> // -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-

Reply:
Done, follow:
> // -- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t --